app/src/main/java/com/topjohnwu/magisk/core/tasks/MagiskInstaller.kt

```diff
@@ -160,6 +160,37 @@ abstract class MagiskInstallImpl protected constructor(
         return true
     }
 
+    private fun addond(): Boolean {
+        console.add("- Adding addon.d survival script")
+
+        val addond = "/system/addon.d"
+
+        if (!SuFile("$addond/99-magisk.sh").exists()) {
+            console.add("! Lygisk was not installed from zip")
+            console.add("! and did not create addon.d files")
+            return false
```

Comment:
return true or false here, or don't return at all and continue anyway?

Reply:
```suggestion
            return true
```
I'd say just spit a warning and continue.
This way the update will work even when Lygisk was not installed from recovery.

Reply:
```suggestion
```
but if addond doesnt exist it would install it anyways wouldn't it?
This way a user would have the Lygisk benefit anyways after one update.
I guess we would have to try this out.

Reply:
Ok, I'll remove the check